### PR TITLE
Adds generic responder and transformer. Updates dependencies.

### PR DESCRIPTION
**Pull request by golonix**

### CODE REVIEW

**Comment by golonix** [comment]
@wlucjan when you think it is ok, please reassign this PR to @fojuth

File: src/Fractal/Transformer.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * @param string $className
+     * @return TransformerAbstract
+     * @throws \InvalidArgumentException
+     */
+    protected function getTransformer($className)
+    {
+        if (true === array_key_exists($className, $this->map)
```
Comment (wlucjan) [inline src/Fractal/Transformer.php]: I suggest pulling this out into protected method like assertTransformerIsDefined and make it more as a guard - so check if false === $this->assertTransformerIsDefined and inside of conditional throw exception, and move return outside of conditional

File: src/Fractal/Transformer.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    /**
+     * @param string $className
+     * @return string
+     * @throws \InvalidArgumentException
+     */
+    protected function getResourceName($className)
+    {
+        if (true === array_key_exists($className, $this->map)
```
Comment (wlucjan) [inline src/Fractal/Transformer.php]: I suggest pulling this out into protected method like assertResourceNameIsDefined and make it more as a guard - so check if false === $this->assertResourceNameIsDefined and inside of conditional throw exception, and move return outside of conditional

File: tests/Fractal/TransformerTest.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                        'links' => []
+                    ]
+                ]
+            ],
+            $scope->toArray()
+        );
+    }
+
+    /** @test */
+    public function it_should_transform_item_with_includes()
```
Comment (wlucjan) [inline tests/Fractal/TransformerTest.php]: I would add a test for case without parseIncludes() called too

**Comment by wlucjan** [comment]
no more remarks from me; assigned to @fojuth

File: config/adr.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    | Transforming mechanism is using Fractal package by PHP League.
+    |
+    */
+    'transformers' => [
+        'EntityClass' => [
+            'transformer' => 'TransformerClass',
+            'resource' => 'ResourceName'
+        ],
+    ]
+];
```
Comment (fojuth) [inline config/adr.php]: Missing newline.

**Pull request merged**